packages/apollo-forest-run/src/diff/diffObject.ts

```diff
@@ -508,9 +518,20 @@ function diffCompositeListLayout(
       firstDirtyIndex = i;
       break;
     }
+    unusedBaseIndixes.delete(i);
   }
   // Fast-path: no layout difference found
   if (firstDirtyIndex === -1) {
+    for (const index of unusedBaseIndixes) {
+      listContext.push(
+        LayoutChange.createItemRemoved(
```

Comment:
This is also somewhat counter intuitive, because there `createItemRemoved` may return undefined when history is disabled. I think it is valuable to keep added/removed info in the difference itself (even when history is disabled).

I.e. I propose to change CompositeListLayoutDifference type to be an object:

```
type CompositeListLayoutDifference = {
  newLayout: CompositeListLayout,
  itemsRemoved: CompositeListLayoutChangeItemRemoved[],
}
```